src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/OutputFactory.java

```diff
@@ -427,9 +383,28 @@ public Void visitVariable(Declaration.Variable tree, Declaration parent) {
             return null;
         }
 
+        Class<?> clazz = getJavaType(type);
+        if (clazz == null) {
+            String name = parent != null? parent.name() + "." : "";
+            name += fieldName;
+            warn("skipping " + name + " because of unsupported type usage");
+            return null;
+        }
+
+
+        VarInfo varInfo = VarInfo.ofVar(clazz, layout);
         Type.Function func = getAsFunctionPointer(type);
+        String fiName;
         if (func != null) {
-            generateFunctionalInterface(func, fieldName);
+            fiName = generateFunctionalInterface(func, fieldName);
+            if (fiName != null) {
+                varInfo = VarInfo.ofFunctionalPointerVar(clazz, layout, fiName);
+            }
+        } else {
+            Optional<String> funcTypedef = getAsFunctionPointerTypedef(type);
+            if (funcTypedef.isPresent()) {
+                varInfo = VarInfo.ofFunctionalPointerVar(clazz, layout, Utils.javaSafeIdentifier(funcTypedef.get()));
+            }
         }
```

Comment:
I see what you mean here. I guess we expect these functional interface classes to be generated when visiting the typedef?

Reply:
yeah - not great - as the functional interface name might differ significantly from the typedef name in case of clashes...

Reply:
Can we rely on typedef names being unique? If so, we could build a map String -> FunctionBuilder when we construct a functional interface for a typedef.

In a header file this should perhaps always be the case - but I've seen that you can have suspicious stuff like

```
{
typedef int a;
}

{
typedef float a;
}
```
But maybe we shouldn't worry about this with C headers?

Reply:
I'm not sure if typedefs have any scope in C, or if they're just global always. C++ has scoped typedefs, but I think in C every name period is global.

Reply:
Godbolt says I'm wrong though :) The example you give is allowed, and the typedef name's scope is restricted to the block

Reply:
The snippet I typed above compiles (e.g. no error about redefinition of same name, which you do get w/o the braces).

Reply:
C Standard:

An identifier can denote an object; a function; a tag or a member of a structure, union, orenumeration;  a  typedef  name; a  label  name;  a  macro  name;  or  a  macro  parameter. The same identifier can denote different entities at different points in the program.

Reply:
> But that could only happen inside a function body - unless I'm missing something?
Yeah, I think that's correct.

Reply:
Then I think I'll settle on a map from String to FunctionalIterfaceBuilder

Reply:
Actually, I did better and added a map from the delegated type to the functional class name - since both declaration and type support deep equality.

Reply:
That's an even better solution!